hugolib/node.go

```diff
@@ -144,10 +180,147 @@ type URLPath struct {
 	Section   string
 }
 
+func (n *Node) URL() string {
+	return n.addLangPathPrefix(n.URLPath.URL)
+}
+
+func (n *Node) Permalink() string {
+	return permalink(n.URL())
+}
```

Comment:
Yes, I kind of agree ... but ... you will find a lot of stuff that should be done, but I think we should draw some line. I have put a lot of sweat into making better tests, so changes like you suggest should be safer to do. My big motivation for the change above was to get a simple 5 lines template example listing all the translations of a page or a node without all the IsPage and IsHome special cases. I feel that trumps a little code impurity for a while.